pkg/cmd/ollama/ollama.go

```diff
@@ -0,0 +1,278 @@
+// Package ollama is for starting AI/ML model workspaces
+package ollama
+
+import (
+	_ "embed"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/brevdev/brev-cli/pkg/cmd/util"
+	"github.com/brevdev/brev-cli/pkg/config"
+	"github.com/brevdev/brev-cli/pkg/entity"
+	"github.com/brevdev/brev-cli/pkg/store"
+	"github.com/brevdev/brev-cli/pkg/terminal"
+	"github.com/spf13/cobra"
+
+	"github.com/brevdev/brev-cli/pkg/cmd/hello"
+	breverrors "github.com/brevdev/brev-cli/pkg/errors"
+)
+
+var (
+	ollamaLong    = "Start an Ollama server with specified model types"
+	ollamaExample = `
+  brev ollama --model llama3
+	`
+	modelTypes = []string{"llama3"}
+)
+
+//go:embed ollamaverb.yaml
+var verbYaml string
+
+type OllamaStore interface {
+	util.GetWorkspaceByNameOrIDErrStore
+	GetActiveOrganizationOrDefault() (*entity.Organization, error)
+	GetCurrentUser() (*entity.User, error)
+	CreateWorkspace(organizationID string, options *store.CreateWorkspacesOptions) (*entity.Workspace, error)
+	GetWorkspace(workspaceID string) (*entity.Workspace, error)
+	BuildVerbContainer(workspaceID string, verbYaml string) (*store.BuildVerbRes, error)
+	ModifyPublicity(workspace *entity.Workspace, applicationName string, publicity bool) (*entity.Tunnel, error)
+}
+
+func validateModelType(modelType string) bool {
+	for _, v := range modelTypes {
+		if modelType == v {
+			return true
+		}
+	}
+	return false
+}
+
+func NewCmdOllama(t *terminal.Terminal, ollamaStore OllamaStore) *cobra.Command {
+	var model string
+
+	cmd := &cobra.Command{
+		Use:                   "ollama",
+		DisableFlagsInUseLine: true,
+		Short:                 "Start an Ollama model server",
+		Long:                  ollamaLong,
+		Example:               ollamaExample,
+		Annotations: map[string]string{
+			"quickstart": "",
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if model == "" {
+				return fmt.Errorf("model type must be specified")
+			}
+
+			isValid := validateModelType(model)
+			if !isValid {
+				return fmt.Errorf("invalid model type: %s", model)
+			}
+
+			// Channel to get the result of the network call
+			resultCh := make(chan error)
+			defer close(resultCh) // TODO: im thinking this is needed?
+
+			// Start the network call in a goroutine
+			go func() {
+				err := runOllamaWorkspace(t, model, ollamaStore)
```

Comment:
Lets copy over the the async function from devplane collections. Its really useful for this kind of thing without having to wrangle goroutines manually